### PR TITLE
feat(Divider,Flex,Price,Text): added native components

### PR DESCRIPTION
**Pull request by kashishgrover**
feat(native/Divider): divider component
feat(native/Flex): flex component
feat(native/Price): price component
fix(native/Text):  font family and line height

### CODE REVIEW

File: src/components/Divider/native/Divider.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const Divider = styled.View`
+  borderTopWidth: ${(p) => p.vertical ? 0 : p.thickness};
```
Comment (lakshyaranganath) [inline src/components/Divider/native/Divider.js]: turn camelCase css properties to kebab-case

File: src/components/Divider/native/Divider.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  borderRightWidth: ${(p) => p.vertical ? p.thickness : 0};
```
Comment (lakshyaranganath) [inline src/components/Divider/native/Divider.js]: rename `p` to `props`

File: src/components/Divider/native/Divider.js (hunk 1 above)
Comment (kashishgrover) [reply src/components/Divider/native/Divider.js]: done

File: src/components/Divider/native/Divider.js (hunk 2 above)
Comment (kashishgrover) [reply src/components/Divider/native/Divider.js]: done

File: src/components/Divider/native/Divider.js [hunk 3]
```diff
@@ -2,11 +2,12 @@ import PropTypes from 'prop-types';
...
 
 const Divider = styled.View`
-  borderTopWidth: ${(p) => p.vertical ? 0 : p.thickness};
-  borderRightWidth: ${(p) => p.vertical ? p.thickness : 0};
-  borderColor: ${(p) => p.theme.color[p.color]};
-  borderStyle: ${(p) => p.type};
-  margin: 0;
+  ${(props) => props.thickness ? `border-top-width: ${props.vertical ? 0 : props.thickness};` : ''};
+  ${(props) => props.thickness ? `border-right-width: ${props.vertical ? props.thickness : 0};` : ''};
+  ${(props) => props.thickness ? `border-color: ${props.theme.color[props.color]};` : ''};
```
Comment (kashishgrover) [inline src/components/Divider/native/Divider.js]: oh shit :P

File: src/native.js [hunk 4]
```diff
@@ -4,4 +4,8 @@ export Form from './components/Form';
 export Spacer from './components/Spacer';
 export Text from './components/Text/native';
 export TextInput from './components/TextInput/native';
+export Divider from './components/Divider/native';
+export Price from './components/Price/native';
+export Flex from './components/Flex/native';
+
```
Comment (lakshyaranganath) [inline src/native.js]: remove this empty line

File: src/components/Text/native/Text.js [hunk 5]
```diff
@@ -1,29 +1,46 @@
...
-    return `font-family: ${p.theme.fontFamily[p.weight]}`;
+    return `font-family: ${props.theme.fontFamily[props.weight]}`;
   }}
   `;
 
 Text.propTypes = {
   color: PropTypes.string,
   size: PropTypes.string,
   weight: PropTypes.string,
+  lineHeight: PropTypes.number,
```
Comment (lakshyaranganath) [inline src/components/Text/native/Text.js]: Don't allow lineHeight customization, our style guide rules are:
if it is a single line of text, lineHeight should be default
if it is a paragaph of text, lineHeight should be set to 1.5

**Comment by kashishgrover** [comment]
Too many issues with this. Closing it for now; will raise PR's for each of the components separately 🙂

**Pull request closed without merging**